src/AutoRest.CSharp.V3/Generation/Writers/ClientWriter.cs

```diff
@@ -365,7 +366,11 @@ private void WriteParameterNullChecks(CodeWriter writer, IEnumerable<Parameter>
                     }
                 }
             }
-            writer.Line();
+
+            if (parameters.Count > 0)
```

Comment:
Any() and keep it enumerable?

Reply:
Multiple enumerations of it would occur. Using IReadOnlyCollection is more accurate since we only ever provide input as an array anyway.

Reply:
Nope, `Any` knows to downcast to IList before doing the enumeration.

Reply:
https://github.com/dotnet/runtime/blob/master/src/libraries/System.Linq/src/System/Linq/AnyAll.cs#L12

Reply:
Right, if the input is IList. But, having IEnumerable as the parameter type says it can be anything that implements IEnumerable. So it still could have multiple enumerations (that's what I meant in my first comment). At least with IReadOnlyCollection, we are guaranteeing that we are only reading the values and have access to `.Count` naturally. I believe that is the point of that interface, right? I'd rather guaranteed non-multiple enumerations of the collection (for anything, not just this method). It is less issue prone (performance) that way.

Reply:
Don't want to spend to much time on this, just know that .Any is optimized not to start enumeration when underlying structure has length.

Reply:
Yeah, I didn't know that explicitly. I use `Any` in most cases. I'll change it to Any here, since with IReadOnlyCollection, it is guaranteed to use `.Count`.